Macro fps support

arikheinss opened the pull request:
Hello.

I rewrote the `_animate` function, so that the animation macros `@gif`, `@animate`  and `@apng` support an extra fps-parameter. 
This includes some restructuring which could make possible future parameter additions somewhat simpler. 

I also added Documentation for these macros to properly communicate this functionality.

t-bltg commented:
Thanks for this PR. You might want to read https://docs.juliaplots.org/stable/contributing/#Write-code,-and-format.

It would be great to add some tests in https://github.com/JuliaPlots/Plots.jl/blob/master/test/test_animations.jl.

arikheinss commented on src/animation.jl:
````diff
@@ -275,6 +289,11 @@ Example:
     push!(p, 1, sin(x))
   end
 ```
+This macro supports additional parameters, that may be added after the main loop body.
+- Add `fps=n` with positive Integer n, to specify the desired frames per second.
+- Add `every n` with positive Integer n, to take only one frame every nth iteration.
+- Add `when <cond>` where `<cond>` is an Expression resulting in a Boolean, to take a 
+    frame only when `<cond>` returns `true`. Is incompatible with `every`.
 """
 macro apng(forloop::Expr, args...)
````
I tried this, with 
```julia
let moreDocumentation = """This macro supports ... """
    @doc string(@doc @gif)*moreDocumentation :(@gif)
    @doc string(@doc @apng)*moreDocumentation :(@apng)
    @doc string(@doc @animate)*moreDocumentation :(@animate)
end
```
but instead of extending the documentation it added a whole new one. It was not very pretty.

arikheinss commented:
Short FYI, this is my first real Pull request ever, so I don't quite know what I am doing, but I just got an automated email pointing me to [this](https://docs.juliaplots.org/stable/contributing/) page, which should help.
I will see whether I can improve the request over the weekend.

arikheinss commented:
Greetings

Over the weekend I cleaned up and formatted the code. I also added some tests. 

BTW, since the macro just passes some data to the `animate` function, and my commit enables it to read in kwargs, this might be an opportunity to think about which parameters `@gif` should be allowed to pass down (maybe `loop` or `variable_palette`?). 
If any new kwargs are to be allowed they just have to be appended in the tuple at `if lhs in (:fps,)`. If _any_ kwarg should be allowed the check should just be removed.

BeastyBlacksmith commented on src/animation.jl:
```diff
@@ -199,26 +199,36 @@ function _animate(forloop::Expr, args...; type::Symbol = :none)
+        elseif arg isa Expr && arg.head == Symbol("=")
+            #specification of type <kwarg> = <spec>
+            lhs, rhs = arg.args
+            if lhs in (:fps,)
```
I think its fine to just pass unconditionally
